pkg/cache/snapshot.go

```diff
@@ -144,6 +145,18 @@ func (c *ClusterQueue) snapshot() *ClusterQueue {
 		// Shallow copy is enough.
 		cc.Workloads[k] = v
 	}
+
+	if features.Enabled(features.LendingLimit) {
+		cc.GuaranteedQuota = make(FlavorResourceQuantities, len(c.GuaranteedQuota))
```

Comment:
You are replacing `cc.GuarnteedQuota` in `update`, so a shallow copy should be ok here.

Reply:
Done.